server/api/user/update/controller.test.js

```diff
@@ -0,0 +1,355 @@
+  });
+
+  test('updates user with new image if picture url is provided', async () => {
+    const newPictureURL = 'https://www.acme.com/pictures/v2/coyote.png';
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        picture: newPictureURL,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        picture: newPictureURL,
+        fullName: user.fullName,
+        username: user.username,
+        emails: expect.any(Array),
+        orgs: expect.any(Array),
+      },
+    });
+  });
+  test('removes image from user if picture property is null', async () => {
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        picture: null,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        picture: null,
+        fullName: user.fullName,
+        username: user.username,
+        emails: expect.any(Array),
+        orgs: expect.any(Array),
+      },
+    });
+  });
+
+  test('updates user and returns expected response', async () => {
+    const newPictureURL = 'https://www.acme.com/pictures/v2/coyote.png';
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        fullName: 'new Coyotee',
+        username: 'alibaba',
+        emails: [{
+          address: 'coyote@acme.com',
+          isPrimary: true,
+        }, {
+          address: 'coyote-new@acme.com',
+        }],
+        picture: newPictureURL,
+      });
+
+    expect(res.status).toBe(200);
+    expect(res.body).toMatchObject({
+      ok: true,
+      user: {
+        id: user.id,
+        fullName: 'new Coyotee',
+        username: 'alibaba',
+        emails: expect.arrayContaining([{
+          address: expect.any(String),
+          isPrimary: expect.any(Boolean),
+        }]),
+        picture: newPictureURL,
+      },
+    });
+  });
+
+  test.only('updates user password', async () => {
+    const newPassword = 'thi$-$s-$af3r';
+    const res = await request(server)
+      .post('/api/user.update')
+      .set('Authorization', `Bearer ${session.accessToken}`)
+      .send({
+        id: user.id,
+        password: newPassword,
+      });
+
+    expect(res.status).toBe(200);
+    const newSession = await createSession(ctx, {
+      username: 'wile',
+      password: newPassword,
+    });
+    // destroying the session before asserting in case of test failing
+    await destroySession(ctx, newSession);
+    expect(newSession).toMatchObject({
+      accessToken: expect.any(String),
+      refreshToken: expect.any(String),
+    });
+  });
+});
```

Comment:
Overall, excellent work with tests. They are very elaborate and cover most use case I can think of.

I suggest however we test for the 2 major user flows:

1. Requestor updates their own profile
2. Requestor us superadmin

I would personally go with 2 major `describe` sections under the root `describe`, i.e. one for every user flow.